warn/warn_test.go

```diff
@@ -413,3 +413,34 @@ d += foo + bar
 		},
 		false)
 }
+
+func TestStringIteration(t *testing.T) {
+	checkFindings(t, "string-iteration", `
+s = "foo" + bar
+
+max(s)
+min(s)
+all(s)
+any(s)
+reversed(s)
+zip(s, a, b)
+zip(a, s)
+
+[foo(x) for x in s]
```

Comment:
nit. add also a test that doesn't show a warning (e.g. argument is a list)

Reply:
Done.